lib/commands/config.js

```diff
@@ -73,6 +74,10 @@ const publicVar = k => {
       }
     }
   }
+  // Hide proxy url if it contains basic auth
+  if (k === 'proxy') {
```

Comment:
This is confusing to me.  This function is a true/false test of whether or not npm should hide the value altogether.  This make the function try to do two things, and I don't think it's doing it correctly down below.   Or if it is, I don't understand how.

Reply:
Ooooh I see, it's comparing it to the redacted value.  If it has private info it won't return.

Can we redact this and display it instead?  Showing it sometimes and not others seems weird to me.

Reply:
Took another attempt at this and made a new function in addition to `publicVar` that determines how to display a field. It will treat redacted and protected fields the same (so no individual getting, excluding from json) except in the case of `npm config ls` in which case it will show the redacted string like `http://u:***@registry.com` instead of `(protected)`.

Reply:
> If we show a redacted value, do we treat "redacted" vs "protected" differently in the different forms of output

Yes.  Some fields we know are always not to be shown, and also unlikely to be caught by our very limited redaction attempts.  Those can not be trusted to be redacted and will always need to show `protected`

Reply:
👍🏼 That's what I did in my latest commit.